src/hffs/fs.py

```diff
@@ -88,58 +86,84 @@ class HfFileSystem(fsspec.AbstractFileSystem):
         )
 
-    def _dircache_from_repo_info(self):
-        repo_info = self._api.repo_info(
-            self.repo_id, revision=self.revision, repo_type=self.repo_type, token=self.token, files_metadata=True
-        )
+    def _resolve_repo_id(self, path: str) -> Tuple[str, str, str]:
+        path = self._strip_protocol(path)
+        if path.split("/")[0] in huggingface_hub.constants.REPO_TYPES_MAPPING:
+            if "/" not in path:
+                raise NotImplementedError("Listing repositories is not implemented.")
+            repo_type, path = path.split("/", 1)
+            repo_type = huggingface_hub.constants.REPO_TYPES_MAPPING[repo_type]
+        else:
+            repo_type = None
+        if path.count("/") > 0:
+            try:
+                namespace, repo_name, *parts_in_repo = path.split("/")
+                repo_id_with_namespace = f"{namespace}/{repo_name}"
+                self._api.repo_info(repo_id_with_namespace, repo_type=repo_type)
+                repo_id = repo_id_with_namespace
+            except huggingface_hub.utils.RepositoryNotFoundError:
+                try:
+                    repo_id_without_namespace, *parts_in_repo = path.split("/")
+                    self._api.repo_info(repo_id_without_namespace, repo_type=repo_type)
+                    repo_id = repo_id_without_namespace
+                except huggingface_hub.utils.RepositoryNotFoundError:
+                    raise FileNotFoundError(f"No such repository: '{repo_id_with_namespace}'")
+        else:
+            try:
+                repo_id, parts_in_repo = path, []
+                self._api.repo_info(path, repo_type=repo_type)
+            except huggingface_hub.utils.RepositoryNotFoundError:
+                raise FileNotFoundError(f"No such repository: '{repo_id}'")
+        path_in_repo = "/".join(parts_in_repo)
+        return repo_type, repo_id, path_in_repo
+
+    def _dircache_from_repo_info(self, path: str):
+        repo_type, repo_id, _ = self._resolve_repo_id(path)
+        repo_info = self._api.repo_info(repo_id, revision=self.revision, repo_type=repo_type, files_metadata=True)
         child_dirs = collections.defaultdict(set)
         for repo_file in repo_info.siblings:
+            hf_path = (
+                huggingface_hub.constants.REPO_TYPES_URL_PREFIXES.get(repo_type, "")
+                + repo_id
+                + "/"
+                + repo_file.rfilename
+            )
             child = {
-                "name": repo_file.rfilename,
+                "name": hf_path,
                 "size": repo_file.size,
                 "type": "file",
                 # extra metadata for files
                 "blob_id": repo_file.blob_id,
                 "lfs": repo_file.lfs,
             }
-            parents = list(PurePosixPath(repo_file.rfilename).parents)[:-1] + [self.root_marker]
+            parents = list(PurePosixPath(hf_path).parents)[:-1] + [self.root_marker]
```

Comment:
I think we need to do the following here to get the correct behavior:
```python
repo_sibling_prefix = huggingface_hub.constants.REPO_TYPES_URL_PREFIXES.get(repo_type, "") + repo_id + "/"
    for repo_file in repo_info.siblings:
        hf_path = repo_sibling_prefix + repo_file.rfilename
        child = {
             "name": hf_path,
             "size": repo_file.size,
             "type": "file",
              # extra metadata for files
              "blob_id": repo_file.blob_id,
              "lfs": repo_file.lfs,
         }
         parents = list(PurePosixPath(hf_path).parents)[:-repo_sibling_prefix.count("/")]
```

Reply:
Thansk for the fix !